control_toolbox/include/control_toolbox/filter_traits.hpp

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2023, Stogl Robotics Consulting UG (haftungsbeschränkt)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CONTROL_TOOLBOX__FILTER_TRAITS_HPP_
+#define CONTROL_TOOLBOX__FILTER_TRAITS_HPP_
+
+#define EIGEN_INITIALIZE_MATRICES_BY_NAN
+
+#include <Eigen/Dense>
+#include <algorithm>
+#include <cmath>
+#include <limits>
+#include <vector>
+
+#include "geometry_msgs/msg/wrench_stamped.hpp"
+
+namespace control_toolbox
+{
+
+template <typename T>
+struct FilterTraits;
+
+// Wrapper around std::vector<double> to be used as
+// the std::vector<double> StorageType specialization.
+// This is a workaround for the fact that
+// std::vector<double>'s operator* and operator+ cannot be overloaded.
+struct FilterVector
+{
+  std::vector<double> data;
+
+  FilterVector() = default;
+
+  explicit FilterVector(const std::vector<double> & vec) : data(vec) {}
+
+  explicit FilterVector(size_t size, double initial_value = 0.0) : data(size, initial_value) {}
+
+  FilterVector operator*(double scalar) const
+  {
+    FilterVector result = *this;
+    for (auto & val : result.data)
+    {
+      val *= scalar;
+    }
+    return result;
+  }
+
+  FilterVector operator+(const FilterVector & other) const
+  {
+    assert(data.size() == other.data.size() && "Vectors must be of the same size for addition");
+    FilterVector result = *this;
+    for (size_t i = 0; i < data.size(); ++i)
+    {
+      result.data[i] += other.data[i];
+    }
+    return result;
+  }
+
+  size_t size() const { return data.size(); }
+};
+
+// Enable scalar * FilterVector
+inline FilterVector operator*(double scalar, const FilterVector & vec) { return vec * scalar; }
+
+template <typename T>
+struct FilterTraits
+{
+  using StorageType = T;
+
+  static void initialize(StorageType & storage)
+  {
+    storage = T{std::numeric_limits<T>::quiet_NaN()};
+  }
+
+  static bool is_nan(const StorageType & storage) { return std::isnan(storage); }
+
+  static bool is_infinite(const StorageType & storage) { return !std::isfinite(storage); }
```

Comment:
i'd prefer to use the `is_finite` instead of `is_infinite`, as we are used to this from std